pkg/controller/core/localqueue_controller.go

```diff
@@ -108,6 +107,17 @@ func (r *LocalQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	log := ctrl.LoggerFrom(ctx)
 	log.V(2).Info("Reconcile LocalQueue")
 
+	// Check if metrics should no longer be recorded in case the namespace labels have changed which wouldn't
+	// be captured by the Update logic
+	if should, _ := metrics.ShouldReportLocalMetrics(r.client, &queueObj); !should {
+		metrics.ClearLocalQueueMetrics(
+			metrics.LocalQueueReference{
+				Name:      queueObj.Name,
+				Namespace: queueObj.Namespace,
+			},
+		)
+	}
```

Comment:
I'm wondering if we need to watch Namespaces by LqReconciler, and then detecting label changes.
If we observe removal of lq metrics label from namespace, we should clear localQueue in event handlers.

Reply:
Would setting up a watch on Namespaces be an okay thing to do? I opted to capture changes to namespace labels lazily when the LQ reconciler retriggers.